### PR TITLE
seatd: honor systemdSupport attr

### PR DESCRIPTION
**Pull request by ghost**
This package causes the `seatd` derivation to honor the `systemdSupport` attr if present.

The default behavior (i.e. when `systemdSupport` is undefined) is unchanged.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/seatd/default.nix [hunk 1]
```diff
@@ -25,9 +25,11 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ meson ninja pkg-config scdoc ];
 
-  buildInputs = [ systemd ];
+  buildInputs = lib.optionals systemdSupport [ systemd ];
 
-  mesonFlags = [ "-Dlibseat-logind=systemd" "-Dlibseat-builtin=enabled" ];
+  mesonFlags = if systemdSupport
+               then [ "-Dlibseat-logind=systemd" "-Dlibseat-builtin=enabled" ]
+               else [ "-Dserver=enabled" "-Dseatd=enabled" "-Dlogind=disabled" "-Dauto_features=auto" ];
```
Comment (alyssais) [inline pkgs/applications/misc/seatd/default.nix:32]: Do we really need a completely different set of build flags just to toggle systemd support?

Comment (a-m-joseph) [reply pkgs/applications/misc/seatd/default.nix:32]: Sorry about that, let me clean it up.

Comment (a-m-joseph) [reply pkgs/applications/misc/seatd/default.nix]: Please see newly-pushed patch.

I've been carrying this patch for a while (through the libseat->seatd rename).  It looks like at some point seatd renamed their options, and meson silently allows options it doesn't understand (ugh).  I pruned the inert options.

**Comment by a-m-joseph** [comment]
Latest force-push collapses the PR into a single commit and adds a commit message.

No other changes.

**Comment by a-m-joseph** [comment]
Rebased.

**Comment by a-m-joseph** [comment]
@alyssais [alyssais](https://github.com/alyssais), did my changes address your concerns?  Do you have any others?

Although #160972 has been merged this PR is also needed in order to keep sway from dragging in systemd via seatd.

**Comment by SuperSandro2000** [comment]
@a-m-joseph feel free to give me a ping if you need a merge or some PR is stuck. I can usually help out.

Currently ofborg fails to build the package because 

```
builder for '/nix/store/m4nrv2ahirhfmxfw9990kv7nw3kswv0d-seatd-0.6.4.drv' failed to produce output path '/nix/store/xdzhdpcnmkrcs33g867s9njvpjb1n3hq-seatd-0.6.4-bin'
```

I think we need to make the bin output depended on systemd but I think this is a bug because ``result-bin/bin/seatd`` and ``result-bin/bin/seatd-launch`` are not build.

**Comment by a-m-joseph** [comment]
@ofborg eval

**Comment by a-m-joseph** [comment]
> I think we need to make the bin output depended on systemd but I think this is a bug because `result-bin/bin/seatd` and `result-bin/bin/seatd-launch` are not build.

I'm unable to reproduce @ofborg's problem.

When I check out 89cc858de3b8e58c5a3f75ec177d43e4f92797ea and run `nix-build . -A seatd` I get:

```
$ tree /nix/store/l020sd4vwsjx71m0hdxb6m0sjrdpy95d-seatd-0.6.4-bin
/nix/store/l020sd4vwsjx71m0hdxb6m0sjrdpy95d-seatd-0.6.4-bin
`-- bin
    |-- seatd
    `-- seatd-launch

1 directory, 2 files
```

I think this might be an @ofborg bug.  @Grahamc, is there some way for me to get the exact git commit-hash and exact `nix-build` command line that was used in a build?  I clicked the "View more details on OfBorg" but couldn't find these anywhere.

**Comment by SuperSandro2000** [comment]
> I think this might be an https://github.com/ofborg?type=source bug.

I highly doubt that. Ofborg merges the branch before building. Can you try rebasing your branch on master and try again.

**Comment by a-m-joseph** [comment]
> Can you try rebasing your branch on master and try again.

Ah yeah, looks like a2122813921a74f1c6f04f113b07f89484376765 on 21-Feb changed how the meson options work.

**Comment by a-m-joseph** [comment]
Yeah, seatd commit https://git.sr.ht/~kennylevinsen/seatd/commit/2204db5531ed16bf32f969645e7177f6118f8a8e changed the names of all the options, and then that commit got included in the 0.6.4 release.

I will adjust for this.

**Comment by a-m-joseph** [comment]
Okay, should be working again now.

**Comment by a-m-joseph** [comment]
Hrm, seatd-0.6.4 also improved the meson feature detection, so I was able to rewrite the `mesonFlags` to make them no longer dependent on the `systemdSupport` variable.  The new seatd is able to figure out the correct flags simply from what's visble in the build environment (`buildInputs`) when you pass `-Dauto_features=auto`.

I assume that this arrangement is preferred for nixpkgs.

The one option that has to be set manually is `-Dlibseat-builtin=enabled`.  I think that allows the old pre-libseat behavior for sway, where you could make `bin/sway` setuid-root and it would `fork()` itself into two processes, one of which does the job of `seatd`.  In any case, the `-Dlibseat-builtin=enabled` setting predates my commit -- it was added in d5bcba77a36f7266ee234fabddb6f828c681363c, so I made sure not to disturb it.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@alyssais what do you think about this?

**Comment by jtojnar** [comment]
> The new seatd is able to figure out the correct flags simply from what's visble in the build environment (`buildInputs`) when you pass `-Dauto_features=auto`.
> 
> I assume that this arrangement is preferred for nixpkgs.

We are passing `-Dauto_features=enabled` intentionally, to make builds fail loudly when something is not as expected:

https://github.com/NixOS/nixpkgs/blob/cd8d9556208b9f31c3e8695acd13755d206fa849/pkgs/development/tools/build-managers/meson/setup-hook.sh#L15

**Comment by a-m-joseph** [comment]
> > The new seatd is able to figure out the correct flags simply from what's visble in the build environment (`buildInputs`) when you pass `-Dauto_features=auto`.
> > I assume that this arrangement is preferred for nixpkgs.
> 
> We are passing `-Dauto_features=enabled` intentionally, to make builds fail loudly when something is not as expected:
> 
> https://github.com/NixOS/nixpkgs/blob/cd8d9556208b9f31c3e8695acd13755d206fa849/pkgs/development/tools/build-managers/meson/setup-hook.sh#L15

Okay, I have pushed an additional commit which reverts the use of `-Dauto_features=auto`.

**Comment by a-m-joseph** [comment]
Ping...

**Pull request merged**